src/front/glsl/builtins.rs

```diff
@@ -1429,88 +1400,113 @@ impl MacroCall {
-            }
-            MacroCall::TextureProj => {
-                let level = args
-                    .get(2)
-                    .copied()
-                    .map_or(SampleLevel::Auto, SampleLevel::Bias);
-                let size = match *parser.resolve_type(ctx, args[1], meta)? {
-                    TypeInner::Vector { size, .. } => size,
-                    _ => {
-                        return Err(Error {
-                            kind: ErrorKind::SemanticError("Bad call to textureProj".into()),
-                            meta,
-                        })
+
+                    TextureLevelType::Grad(false) => {
+                        num_args += 2;
+                        SampleLevel::Gradient {
+                            x: args[2],
+                            y: args[3],
+                        }
                     }
                 };
-                let base = args[1];
-                let mut right = ctx.add_expression(
-                    Expression::AccessIndex {
-                        base,
-                        index: size as u32 - 1,
-                    },
-                    SourceMetadata::none(),
-                    body,
-                );
-                let left = if let VectorSize::Bi = size {
-                    ctx.add_expression(
-                        Expression::AccessIndex { base, index: 0 },
-                        SourceMetadata::none(),
-                        body,
-                    )
-                } else {
-                    let size = match size {
-                        VectorSize::Tri => VectorSize::Bi,
-                        _ => VectorSize::Tri,
-                    };
-                    right = ctx.add_expression(
-                        Expression::Splat { size, value: right },
-                        SourceMetadata::none(),
-                        body,
-                    );
-                    ctx.vector_resize(size, base, SourceMetadata::none(), body)
+
+                let texture_offset = match offset {
+                    true => {
+                        let offset_arg = args[num_args];
+                        num_args += 1;
+                        Some(parser.solve_constant(ctx, offset_arg, meta)?)
+                    }
+                    false => None
                 };
-                let coords = ctx.add_expression(
-                    Expression::Binary {
-                        op: BinaryOperator::Divide,
-                        left,
-                        right,
-                    },
-                    SourceMetadata::none(),
-                    body,
-                );
-                let comps = parser.coordinate_components(ctx, args[0], coords, meta, body)?;
-                texture_call(ctx, args[0], level, comps, body, meta)
+
+                // Now go back and look for optional bias arg (if available)
+                if level_type == TextureLevelType::None {
```

Comment:
nit: `if let`

Reply:
I don't quite understand what this is asking for?

Reply:
Use `if let TextureLevelType::None = level_type`

Reply:
Ah, thank you. Fixed.